shared/bundle_analysis/models.py

```diff
@@ -0,0 +1,169 @@
+from enum import Enum
+from typing import List
+
+from sqlalchemy import Column, ForeignKey, Table, create_engine, types
+from sqlalchemy.orm import Mapped
+from sqlalchemy.orm import Session as DbSession
+from sqlalchemy.orm import backref, declarative_base, relationship, sessionmaker
+
+SCHEMA = """
+create table sessions (
+    id integer primary key,
+    info text not null
+);
+
+create table metadata (
+    key text primary key,
+    value text not null
+);
+
+create table assets (
+    id integer primary key,
+    session_id integer not null,
+    name text not null,
+    size integer not null,
+    foreign key (session_id) references sessions (id)
+);
+
+create index assets_session_id on assets (session_id);
+create index assets_name_index on assets (name);
+
+create table chunks (
+    id integer primary key,
+    external_id text not null,
+    entry boolean not null,
+    initial boolean not null,
+    unique (external_id)
+);
+
+create table assets_chunks (
+    asset_id integer not null,
+    chunk_id integer not null,
+    primary key (asset_id, chunk_id),
+    foreign key (asset_id) references assets (id),
+    foreign key (chunk_id) references chunks (id)
+);
+
+create table modules (
+    id integer primary key,
+    name text not null,
+    size integer not null
+);
+
+create table chunks_modules (
+    chunk_id integer not null,
+    module_id integer not null,
+    primary key (chunk_id, module_id),
+    foreign key (chunk_id) references chunks (id),
+    foreign key (module_id) references modules (id)
+);
+"""
+
+SCHEMA_VERSION = 1
+
+Base = declarative_base()
+
+
+def get_db_session(path: str) -> DbSession:
+    engine = create_engine(f"sqlite:///{path}")
+    Session = sessionmaker()
+    Session.configure(bind=engine)
+    session = Session()
+    return session
+
+
+assets_chunks = Table(
```

Comment:
Very crafty the way you declared the tables that glue together N:N relationships without having to create new models just for that. I like it.

However it's not immediately obvious why these declarations are different from the rest (at least not to me), and that you don't have to use these as models.

I think a comment before the `Table` declarations explaining what they do and that it doesn't have to be a model in itself but needs to exist in the DB would be helpful here.